Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3255,9 +3255,12 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     m_backgroundWebView.get()._remoteInspectionNameOverride = backgroundWebViewInspectionName();
     m_backgroundContentLoadError = nil;
 
+    auto backgroundPage = m_backgroundWebView.get()._page;
+    Ref process = backgroundPage->protectedLegacyMainFrameProcess();
+    m_activityForBackgroundWebView = process->throttler().foregroundActivity("Web extension activity"_s);
```

Comment:
Seems fishy to have a foreground activity no matter what? Is the WebExtension process always separate from page webProcesses? Normally, I'd expect the WebExtension process's assertion to be derived from the pages it serves. Holding a foreground assertion unconditionally on the extension process seems like it could easily regress power.

Reply:
Unfortunately, a background page does not serve just one tab, it gets events from many different sources. We could hold and release the activity when we are delivering those events, that likely is a better model than always holding it.

I'll be happy to do that work, so we hold the assertion when delivering events to the background page.

I think we should decline this PR in the meantime.